scalding-core/src/main/scala/com/twitter/scalding/Execution.scala

```diff
@@ -475,12 +481,29 @@ object Execution {
    */
 
   private trait ToWrite {
+    def setExtraConfig(conf: Config): Config = Config.empty
+
     def write(config: Config, flowDef: FlowDef, mode: Mode): Unit
   }
+
   private case class SimpleWrite[T](pipe: TypedPipe[T], sink: TypedSink[T]) extends ToWrite {
+
+    var extraConfig = Config.empty
```

Comment:
So in general you've a mutable API, this is a no-no in most of our settings. Any call on an execution should return a new execution, not mutate the existing one. Thread safety and other things are all issues once you start down the mutation route, what happens with two competing setExtraConfig's in different threads?